includes/class-stats.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Stats
+ *
+ * @package wp-job-manager
+ */
+
+namespace WP_Job_Manager;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * This class is responsible for initializing all aspects of stats for wpjm.
+ */
+class Stats {
+	use Singleton;
+
+	/**
+	 * Constructor.
+	 */
+	private function __construct() {
+		$this->init();
+	}
+
+	/**
+	 * Do initialization of all the things needed for stats.
+	 */
+	public function init() {
+		$this->initialize_wpdb();
+	}
+
+	/**
+	 * Initialize the alias for the stats table on the wpdb object.
+	 *
+	 * @return void
+	 */
+	private function initialize_wpdb() {
+		global $wpdb;
+		if ( isset( $wpdb->job_manager_stats ) ) {
+			return;
+		}
+		$wpdb->job_manager_stats = $wpdb->prefix . 'job_manager_stats';
+		$wpdb->tables[]          = 'job_manager_stats';
+	}
+
+	/**
+	 * Migrate the stats table to the latest version.
+	 *
+	 * @return void
+	 */
+	public function migrate() {
+		global $wpdb;
+		$collate = $wpdb->get_charset_collate();
+		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+		\dbDelta(
+			[
+				"CREATE TABLE {$wpdb->job_manager_stats} (
+				`id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
+				`date` date NOT NULL,
+				`post_id` bigint(20) DEFAULT NULL,
+				`name` varchar(255) NOT NULL,
+				`group` varchar(255) DEFAULT '',
+				`count` bigint(20) unsigned not null default 1,
+				PRIMARY KEY (`id`),
+				KEY `name_date_group`  (`name`, `date`, `group`)
```

Comment:
Can we use index instead of `KEY`? They are [synonyms](https://dev.mysql.com/doc/refman/8.3/en/create-table.html#create-table-indexes-keys), but `INDEX` is the most iso-compatible one. (and wp now wants to play better with sqlite). Also nit,  prefixing indices with idx_{table_name}.

```suggestion
				INDEX `idx_wpjm_stats_name_date_group`  (`name`, `date`, `group`)
```

Reply:
Good catch on the index name. Regarding using `INDEX`, for me it is kinda indifferent. WP apparently converts everything to `KEY` [anyway on the `dbDelta` function](https://github.com/WordPress/WordPress/blob/27348531f3fdf048809da977fced3f40b90259de/wp-admin/includes/upgrade.php#L2973-L2974), and it also uses [`KEY` everywhere on the schema](https://github.com/WordPress/WordPress/blob/27348531f3fdf048809da977fced3f40b90259de/wp-admin/includes/schema.php#L56-L237), but oh well...